src/rust/engine/process_execution/src/docker_tests.rs

```diff
@@ -63,15 +63,34 @@ macro_rules! setup_docker {
         );
       }
     }
-
-    docker
   }};
 }
 
+#[tokio::test]
+#[cfg(unix)]
+async fn use_local_runner_if_docker_not_set() {
```

Comment:
This test belongs more properly to a `DockerSwitchCommandRunner`. If written for a leaf-only `docker::CommandRunner`, this test would just look for an error if `Process::docker_image` is not set.

Reply:
Or just land it as-is and I'll write the combinator.

Reply:
Sg. Will merge since this is green and I only have a few hours left to make progress this week before OOO